packages/core/src/VDocument.ts

```diff
@@ -152,4 +153,199 @@ export class VDocument {
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Convert the given nodes to a list with the given `htmlTag`.
+     *
+     * @param nodes
+     * @param htmlTag
+     */
+    _convertToList(nodes: Array<VNode>, htmlTag: string): void {
+        // Lists cannot have atomic nodes as direct children.
+        nodes = Array.from(new Set(nodes.map(node => (node.atomic ? node.parent : node))));
+
+        // If the first and last nodes are within lists, we need to split those
+        // lists.
+        const duplicatedNodes = [];
+        const first = nodes[0];
+        const firstList = this._listAncestor(first);
+        if (firstList) {
+            let child = first;
+            first.ancestor(ancestor => {
+                duplicatedNodes.push(ancestor.splitAt(child));
+                child = ancestor;
+                return ancestor === firstList;
+            });
+            if (!firstList.hasChildren()) {
+                firstList.remove();
+            }
+        }
+        const last = nodes[nodes.length - 1];
+        const lastList = this._listAncestor(last);
+        if (lastList) {
+            let child = last;
+            last.ancestor(ancestor => {
+                duplicatedNodes.push(ancestor.splitAt(child));
+                child = ancestor;
+                return ancestor === lastList;
+            });
+            if (!lastList.hasChildren()) {
+                lastList.remove();
+            }
+        }
+
+        // Create and insert the list before the common ancestor to the first
+        // last node, unless the first node is within a list, in which case
+        // insert before the list instead to avoid inserting the new list within
+        // a list.
+        let list = new ListNode(htmlTag);
+        const listAncestor = this._listAncestor(nodes[0]);
+        if (listAncestor) {
+            listAncestor.before(list);
+        } else {
+            const commonAncestor = nodes[0].ancestor(ancestor => {
+                return !!nodes[nodes.length - 1].ancestor(
+                    otherAncestor => otherAncestor === ancestor,
+                );
+            });
+            commonAncestor.insertBefore(
+                list,
+                nodes[0].ancestor(ancestor => ancestor.parent === commonAncestor) || nodes[0],
+            );
+        }
+
+        // Move the nodes to the list
+        withMarkers(() => {
+            // Remove the children of nested lists (they will be moved together
+            // with their parent) and convert nested lists to the new type.
+            const nestedLists = nodes.filter(this._isNestedList.bind(this));
+            nodes = nodes.filter(node => {
+                return !node.ancestor(ancestor => nestedLists.includes(ancestor));
+            });
+            nestedLists.forEach(nestedList => ((nestedList as ListNode).htmlTag = htmlTag));
```

Comment:
Aren't you supposed to change `listType` ? You might want to redefine `htmlTag` as a getter on `listType` or the opposite to keep them synchronized.

Reply:
ListType _is_ a getter so they remain in sync. Now as I said above I think instantiating lists with listType instead of htmlTag might make more sense so I did that locally yesterday evening. Then I had to overload htmlTag as a getter to keep in sync instead of the opposite. This line is thus now `nestedLists.forEach(nestedList => ((nestedList as ListNode).listType = type));`